
  <!-- Tela Editar -->
  <div id="tela_gerenciar" class="container-main container-fluid">
    <div id="tela_gerenciar_listagem" class="tela container-main">
      <button id="" class="amarelo" onclick=""></button>
    </div>

    <div id="tela_gerenciar_formulario" class="tela container-main container-fluid card">
      <h3>Editar</h3>
      <form>
        <label for="edit_titulo" class="form-label">Título do Filme</label>
        <input type="text" id="edit_titulo" class="form-control">


        <label for="formFile" class="form-label">Default file input example</label>
        <input class="form-control" type="file" id="formFile">


        <label for="edit_descricao" class="form-label">Descrição</label>
        <textarea id="edit_descricao" class="form-control" rows="4"></textarea>

        <button class="btn btn-success" onclick="btn_onclick_alteracoes_filme()">Salvar Edição</button>
        <button class="btn btn-secondary" onclick="btn_onclick_limpar()">Limpar</button>
      </form>
    </div>
  </div>

  <script src="js/bootstrap.bundle.min.js"></script>
</body>

</html>


// rota para SELECT
app.get('/api/usuarios', async (req, res)=>{
    try {
        const conexao = await pool.getConnection() 
        const sql = "SELECT * FROM usuarios"
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json(linhas)
            // cosole.log([linhas])


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na busca"})
    }
})

// rota para SELECT com ID
app.get('/api/usuarios/id/:id', async (req, res)=>{
    try {
        const id = req.params.id
        const conexao = await pool.getConnection() 
        const sql = `SELECT * FROM usuarios WHERE id = ${id}`
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json(linhas[0])
            // cosole.log([linhas])


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na busca"})
    }
})

// nome
app.get('/api/usuarios/nome/:nome', async (req, res)=>{
    try {
        const nome = req.params.nome
        const conexao = await pool.getConnection() 
        const sql = `SELECT * FROM usuarios WHERE nome LIKE "%${nome}%"`
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json(linhas)


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na busca"})
    }
})

// email
app.get('/api/usuarios/email/:email', async (req, res)=>{
    try {
        const email = req.params.email
        const conexao = await pool.getConnection() 
        const sql = `SELECT * FROM usuarios WHERE email LIKE "%${email}%"`
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json(linhas)
            // cosole.log([linhas])


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na busca"})
    }
})

app.delete("/api/usuarios/:id", async(req, res)=>{
    try {
        const id_passado = req.params.id
        const conexao = await pool.getConnection() 
        
        const sql = `DELETE FROM usuarios WHERE id = ${id_passado}`
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json({msg: "Registro excluidos!!"})
            // cosole.log([linhas])


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na exclusão"})
    }
})

app.post("/api/usuarios/", async(req,res) =>{
    try {
        const {nome, email} = req.body
        const conexao = await pool.getConnection() 
        
        const sql = `INSERT INTO usuarios (nome,email) VALUE ("${nome}" ,"${email}")`
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json({msg: "Registro gravado"})
            // cosole.log([linhas])


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na exclusão"})
    }
})

// rota para o UPDATE
app.put("/api/usuarios/", async(req,res)=> {
    try {
        const {id, nome, email} = req.body
        const conexao = await pool.getConnection() 
        
        const sql = `UPDATE usuarios SET nome = "${nome}", email ="${email}" WHERE id =${id}`
        const [linhas] = await conexao.execute(sql)
        conexao.release()
        res.json({msg: "Registro gravado"})
            // cosole.log([linhas])


    } catch (error) {
        console.log(`O erro que ocorreu foi: ${error}`)
        res.send(500).json({error:"Deu algum erro na exclusão"})
    }
})